src/zenml/orchestrators/step_launcher.py

```diff
@@ -423,10 +459,10 @@ def _prepare(
                     step_run=step_run,
                     step_source=self._step.spec.source,
                 )
-                if self._step.config.model:
+                if model:
                     orchestrator_utils._link_pipeline_run_to_model_from_context(
                         pipeline_run_id=step_run.pipeline_run_id,
-                        model=self._step.config.model,
+                        model=model,
```

Comment:
Maybe this question is stupid, but do we still need this `ModelVersionPipelineRunRequest`? It seems like this would now just be an `UPDATE PipelineRun` that sets `PipelineRun.configured_model_version_id`?

Reply:
This is the idea I also shared in the general comment below. Let me check if I can get rid of the MV PR links table completely.

Reply:
So the problem with flattening it down at the moment is the following: it is possible to have N:M relationship for Pipeline Runs vs Model Versions. This happens, if I set one Model Version in pipeline decorator and some other in some step decorators. This is not reflected in the state of the Pipeline Run, but can be awkwardly reconstructed back by looking up in Step/Pipeline Runs.
This is in theory doable, but I would not take such a rework in the scope of this ticket.